x-pack/solutions/observability/plugins/observability/server/services/related_dashboards_client.ts

```diff
@@ -48,18 +53,37 @@ export class RelatedDashboardsClient {
         `Alert with id ${this.alertId} not found. Could not fetch related dashboards.`
       );
     }
+    const [suggestedDashboards, linkedDashboards] = await Promise.all([
+      this.fetchSuggestedDashboards(),
+      this.getLinkedDashboards(),
+    ]);
+    const filteredSuggestedDashboards = suggestedDashboards.filter(
+      (suggested) => !linkedDashboards.some((linked) => linked.id === suggested.id)
+    );
+    return {
+      suggestedDashboards: filteredSuggestedDashboards,
+      linkedDashboards,
+    };
+  }
+
+  async fetchSuggestedDashboards(): Promise<SuggestedDashboard[]> {
```

Comment:
```suggestion
  private async fetchSuggestedDashboards(): Promise<SuggestedDashboard[]> {
```

We could optionally encapsulate this function as we only call it internally in `RelatedDashboardsClient`. I do see it's referenced in the test file so if you'd rather keep it public for that purpose to avoid complicating testing I'm ok with that.